### PR TITLE
Create new parameter for checking the image checksum

### LINKED ISSUE #304: Image upload has mismatched file size and checksum
### Terraform Version
Terraform v0.11.2

### Affected Resource(s)
- openstack_images_image_v2

### Terraform Configuration Files
```hcl
resource "openstack_images_image_v2" "image" {
  name             = "PretendName"
  image_source_url = "https://thisisaexample/pretend.ova"
  container_format = "ova"
  disk_format      = "vmdk"
  visibility       = "private"

  properties {
    hw_vif_model       = "VirtualVmxnet3"
    vmware_disktype    = "streamOptimized"
    vmware_adaptertype = "lsiLogic"
  }
}
```

### Debug Output
N/A

### Expected Behavior
I expect Terraform to not error

### Actual Behavior
```
Error: Error applying plan:

1 error(s) occurred:

* module.images.openstack_images_image_v2.image: 1 error(s) occurred:

* openstack_images_image_v2.image: Error waiting for Image: Error wrong size 31457280000 or checksum ""
```

I went ahead and updated the error to be more clear
```golang
		if img.Checksum != checksum || int64(img.SizeBytes) != fileSize {
			return img, fmt.Sprintf("%s", img.Status), fmt.Errorf("Error wrong size: got %v, expected %v and/or Error checksum: got %q, expected %q", img.SizeBytes, fileSize, img.Checksum, checksum)
		}
```

New error:
```
Error: Error applying plan:

1 error(s) occurred:

* module.images.openstack_images_image_v2.image: 1 error(s) occurred:

* openstack_images_image_v2.image: Error waiting for Image: Error wrong size: got 31457280000, expected 1624823808 and/or Error checksum: got "", expected "7fc8a49edba6a11861651a14b0ba4ccf"
```

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

### Important Factoids
We are using Openstack VIO

### PR DESCRIPTION
**Pull request by ChiefAlexander**
fixes #304

### CODE REVIEW

**Comment by jtopjian** [comment]
Thanks for working on this issue :)

I agree with this fix. As well, I agree with setting the default value to `true`.

The only change I would make is to remove the checksum part out of the refreshfunc entirely. Let's make the refreshfun only check the status of the image. Once the image has the correct status, perform a Get here to retrieve the image struct:

https://github.com/ChiefAlexander/terraform-provider-openstack/blob/3d661d7a0dd5ceda9e42a28f5f3fbccb64b56453/openstack/resource_openstack_images_image_v2.go#L259

And then do the check.

Thoughts?

File: openstack/resource_openstack_images_image_v2.go [hunk 1]
```diff
@@ -257,6 +257,15 @@ func resourceImagesImageV2Create(d *schema.ResourceData, meta interface{}) error
 		return fmt.Errorf("Error waiting for Image: %s", err)
 	}
 
+	img, err := images.Get(imageClient, d.Id()).Extract()
+	if err != nil {
+		return CheckDeleted(d, err, "image")
+	}
+
+	if img.Checksum != fileChecksum && d.Get("verify_checksum").(bool) {
```
Comment (jtopjian) [inline openstack/resource_openstack_images_image_v2.go]: nit: let's do:

```go
verifyChecksum := d.Get("verify_checksum").(bool)
if img.Checksum != fileChecksum && verifyChecksum {
```

Having `d.Id()` inline is generally OK. Using `d.Get` isn't (shouldn't) used inline elsewhere.

Comment (ChiefAlexander) [reply openstack/resource_openstack_images_image_v2.go]: That is fine. For any future pr's that I do can you explain why?

**Comment by jtopjian** [comment]
@ChiefAlexander This is looking good.

Looks like the import acceptance test is failing. The new field will need added to `https://github.com/terraform-providers/terraform-provider-openstack/blob/master/openstack/import_openstack_images_image_v2_test.go#L25` in order for the test to pass.

**Comment by ChiefAlexander** [comment]
Awesome, thank you for pointing that out. I was scratching my head on what I needed to update 😆

File: openstack/resource_openstack_images_image_v2.go (hunk 1 above)
Comment (ChiefAlexander) [reply openstack/resource_openstack_images_image_v2.go]: Ah ok. I was thinking that it was a style thing but wasn't sure. Thanks for explaining.

**Comment by jtopjian** [comment]
LGTM - thank you for your work on this :)

**Pull request merged**